### DISCUSSION TITLE
K3s PVs on Synology

### DISCUSSION THREAD

**Discussion by mattkrass** [differentpla-net/differentpla-net.github.io#31] [General]
I saw your recent post about your Longhorn volumes getting blanked out during an upgrade and you mentioned exploring Synology options. I'm using K3s with a Synology hosting all my PVs over NFS, and it's been working quite reliably. I use [this](https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/) to generate paths from claims, and it works well enough. One gotcha to watch out for is that your worker nodes will just hang on pod creation if they don't have an NFS client installed, or the server doesn't allow them to mount the share. Also make sure your server is enabled for mounting subdirectories of shares.

**Comment by rlipscombe** [comment]
Interesting; thanks.
